test_echo.py

```diff
@@ -0,0 +1,30 @@
+# -*- coding: utf-8 -*-
+import subprocess
+
```

Comment:
Using subprocess to execute the client is clever, but not really required.  If you build the client as a Python function and simply call that function from the `if __name__ == "__main__":` block in your client script file, you can import that function and use it directly.